src/core/ext/filters/rbac/rbac_service_config_parser.cc

```diff
@@ -18,587 +18,657 @@
 
 #include "src/core/ext/filters/rbac/rbac_service_config_parser.h"
 
-#include <stdint.h>
-
+#include <cstdint>
 #include <map>
 #include <string>
 
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
-#include "absl/strings/str_cat.h"
-#include "absl/strings/str_format.h"
 #include "absl/types/optional.h"
 
 #include "src/core/lib/channel/channel_args.h"
-#include "src/core/lib/gprpp/status_helper.h"
-#include "src/core/lib/iomgr/error.h"
-#include "src/core/lib/json/json_util.h"
+#include "src/core/lib/json/json_args.h"
+#include "src/core/lib/json/json_object_loader.h"
 #include "src/core/lib/matchers/matchers.h"
-#include "src/core/lib/transport/error_utils.h"
 
 namespace grpc_core {
 
 namespace {
 
-std::string ParseRegexMatcher(const Json::Object& regex_matcher_json,
-                              std::vector<grpc_error_handle>* error_list) {
-  std::string regex;
-  ParseJsonObjectField(regex_matcher_json, "regex", &regex, error_list);
-  return regex;
-}
+struct RbacConfig {
+  struct RbacPolicy {
+    struct Rules {
+      struct Policy {
+        struct CidrRange {
```

Comment:
The deep nesting of these structs makes it hard to reason about this code without tooling (to answer the 'where the heck am I' question)

Suggest either:
1. flattening the structure (pulling the inner structs up to the top level namespace and giving them longer names)
or
2. moving all methods out-of-line below the structure definitions, so that the overall structure can be grokked on ~1 page of code, and the methods carry location information (in the form of `RbacConfig::RbacPolicy::Rules::Policy::CidrRange::JsonLoader()` for instance)

Reply:
Done.  I went with the second option.